### PR TITLE
Fix/minting-with-listing--rmrk2

### LINKED ISSUE #6149: Atributes/Traits not visible in Properties
I minted [this NFT](https://kodadot.xyz/ksm/gallery/18168345-B02B8226FEE3EAA06D-UCF-MELISSA_1-00000001) today and while I was creating it I added 3 attributes. Here they were:
Screenshot 2023-06-01 at 17 11 58

After minting NFT I cannot see the attributes here: (showing No properties)
Screenshot 2023-06-01 at 17 11 41

In detail -  metadata Im able to see them when I open json file
Screenshot 2023-06-01 at 17 14 40

### LINKED ISSUE #6150: Minting with Listing - not working properly
I minted [this NFT](https://kodadot.xyz/ksm/gallery/18168457-B02B8226FEE3EAA06D-UCF-NAMI_2-00000002). 

During minting, I set the price to 0.1 KSM. Like this: (it requires making an additional transaction!!!)
Screenshot 2023-06-01 at 17 18 18

But after signing only ONE transaction, I got this notification:
Screenshot 2023-06-01 at 17 19 44

There was no option to sign the second transaction. My NFT is minted but NOT listed, even though I chose the option to list it for sale. 

Screenshot 2023-06-01 at 17 19 58

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [x] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #6149
- [x] Closes #6150
- [x] :wrench: fix (Unreported issue AFAIK)  - wrong redirect after mint on RMRK2, it redirected to RMRK
- [x] :wrench: fix  - editions didn't work
- [x] :sparkles:  when minting multiple nft's (editions) redirect to collection page instead on single nft page 
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshot 📸

- [x] My fix has changed UI

## Testing
Test by mint+list something on RMRK2

### <samp>🤖 Generated by Copilot at a76663f</samp>

This pull request adds support for the rmrk2 NFT format and improves the user experience of creating and listing NFTs for sale. It updates the `CreateToken.vue` component, the `scheme.ts` service, the `GalleryItemDescription.vue` component, and the `types.ts` composable to handle the new format and the `useTransaction` composable. It also fixes some path and naming issues.

### <samp>🤖 Generated by Copilot at a76663f</samp>

> _To support the rmrk2 format_
> _We had to update some code paths_
> _We changed `properties`_
> _To `meta.attributes`_
> _And imported a new type from minimark_

### CODE REVIEW

File: components/rmrk/Create/CreateToken.vue [hunk 1]
```diff
@@ -271,7 +268,7 @@ export default class CreateToken extends mixins(
             setTimeout(
               () =>
                 this.listForSale(
-                  createdNFTs.value,
+                  createdNFTs.value[0],
```
Comment (Jarsen136) [inline components/rmrk/Create/CreateToken.vue:271]: Would the length of nfts list be greater than one?

File: components/rmrk/service/scheme.ts [hunk 2]
```diff
@@ -265,6 +266,27 @@ export const toNFTId = (
   return `${blocknumber}-${collection}-${instance}-${sn}`
 }
 
+export const toNFTIdV2 = (
```
Comment (vikiival) [inline components/rmrk/service/scheme.ts:269]: Please use utils from `minimark`

File: components/rmrk/Create/CreateToken.vue (hunk 1 above)
Comment (vikiival) [inline components/rmrk/Create/CreateToken.vue:271]: This basically prevents listing of multiple NFTs

**Comment by prury** [comment]
Testing Process(working properly): 

@daiagi thanks for adding a how to test section! ❤️ 

- Signed two transactions, to mint and list.
- Added four properties to the minted NFT:




- listing ok: 





Minted NFT: https://deploy-preview-6156--koda-canary.netlify.app/ksm/gallery/18200581-7EA1DCF47E98A25067-PNKTST-ZOMBIE_GIRL_PUNK-00000005

Extrinsics: 
https://kusama.subscan.io/tx/0xcf4baa03b4e05fe7596066629d28943192b3ca9b3c382122980bdc2679062698
https://kusama.subscan.io/tx/0x22a1fbd32f6dbcef47006e9603e9614b1431dc37b88d664fb1c10f911601b990

File: components/rmrk/Create/CreateToken.vue (hunk 1 above)
Comment (daiagi) [reply components/rmrk/Create/CreateToken.vue:271]: but this page doesn't mint multiple nft's anyways
we (will) have massmint for that

Comment (daiagi) [reply components/rmrk/Create/CreateToken.vue:271]: https://github.com/kodadot/nft-gallery/pull/6156#discussion_r1216477888

File: components/rmrk/service/scheme.ts (hunk 2 above)
Comment (daiagi) [reply components/rmrk/service/scheme.ts:269]: * https://github.com/kodadot/packages/pull/149#issue-1740281659
* https://github.com/kodadot/packages/pull/147#issue-1740277554

**Review by Jarsen136 — APPROVED** [review]

**Comment by roiLeo** [comment]
> Testing Process(working properly):
> 
>  
> 

> Minted NFT: https://deploy-preview-6156--koda-canary.netlify.app/ksm/gallery/18200581-7EA1DCF47E98A25067-PNKTST-ZOMBIE_GIRL_PUNK-00000005

Shouldn't we have to show `key` & `value` instead of `Section` number?

**Comment by daiagi** [comment]
@roiLeo @prury 
resolved

image

**Comment by prury** [comment]
@daiagi really good, ty!

**Comment by daiagi** [comment]
@vikiival  please release new version of minimark

**Comment by vikiival** [comment]
Screenshot 2023-06-15 at 12 39 06

**Comment by yangwao** [comment]
I guess we should wait for this to upcoming release 👀

File: components/rmrk/Create/CreateToken.vue (hunk 1 above)
Comment (vikiival) [reply components/rmrk/Create/CreateToken.vue:271]: Edition cries in the corner

Screenshot 2023-06-16 at 15 47 11

**Review by vikiival — CHANGES_REQUESTED** [review]
Good except it won't list editions

**Comment by vikiival** [comment]
Looks like editions does not work at all

Screenshot 2023-06-16 at 15 52 33

I set edition to `2`.

It should have minted two waifus (only one minted)
And listing failed (set value 0.3)


https://kusama.subscan.io/extrinsic/18382876-5

**Comment by prury** [comment]
in my defense i only understood how editions works inside koda a few days ago
so lets say I've set my collection to max 2 nfts to be minted, i can only mint two NFTS, but their copies inside that collection could be unlimited ?

**Comment by vikiival** [comment]
Editions replicate the currently minted nft 

So If I create collection with max 2 nfts

And i will create nft with two editions i have now collection with 2/2 (minted/max)

However if i have collection with max 2 nfts and i already have one nft in collection

THE Maximum value for edition is 1

**Comment by prury** [comment]
> Editions replicate the currently minted nft 
> 
> So If I create collection with max 2 nfts
> 
> And i will create nft with two editions i have now collection with 2/2 (minted/max)
> 
> However if i have collection with max 2 nfts and i already have one nft in collection
> 
> THE Maximum value for edition is 1

So, if i understood correctly, when creating a collection, when you define your maximum amount of nfts you also define your maximum amount of editions

If its 4 i can mint one nft with 16 editions?

**Comment by daiagi** [comment]
I admit i also didn't know what edition means
and as a side note i think it's a bad name, better be called Replicas / Copies 

@exezbcz  may i change the name?

**Comment by daiagi** [comment]
@vikiival 
> Looks like editions does not work at all

was a pre-existing bug. fixed

> And listing failed

please try again, worked for me twice
https://kusama.subscan.io/extrinsic/18392547-2
https://kusama.subscan.io/extrinsic/18392677-2

image

**Comment by exezbcz** [comment]
@daiagi yes


Copies are good for naming. More descriptive, for sure. I am also thinking about "issuance."

**Comment by daiagi** [comment]
Let me know.
I prefer copies over issuance

**Comment by exezbcz** [comment]
@daiagi oki copies, is it

**Comment by daiagi** [comment]
> ts still showing editions instead of copies.

done

>Now testing if i can go above the limit:

fixed. it will now mint up to the maximum allowed in the collection

> and then I've tried to mint 10 copies of the first NFT (should be 9 max according to what viki wrote(3/3))

i think you misunderstood. if the max of the collection is set to 3 you are suppose to able to mint only up to 3, not 9
https://github.com/kodadot/nft-gallery/pull/6156#issuecomment-1595095928

> IMO, on the user side, we should have a input stating how many nfts/copies the user still has left.

True. but let's push it off to a new issue

**Comment by prury** [comment]
> > ts still showing editions instead of copies.
> 
> done
> 
> > Now testing if i can go above the limit:
> 
> fixed. it will now mint up to the maximum allowed in the collection

ty, will check.

> 
> > and then I've tried to mint 10 copies of the first NFT (should be 9 max according to what viki wrote(3/3))
> 
> i think you misunderstood. if the max of the collection is set to 3 you are suppose to able to mint only up to 3, not 9 [#6156 (comment)](https://github.com/kodadot/nft-gallery/pull/6156#issuecomment-1595095928)

oh, you are correct, i did misunderstood, its way simpler than what i have previously tough, thank you daiagi! 

> 
> > IMO, on the user side, we should have a input stating how many nfts/copies the user still has left.
> 
> True. but let's push it off to a new issue

will do!

**Comment by prury** [comment]
@daiagi can you give me a hand and test it if you are being able to mint without listing with this deployed version? it won't work here for me, the transaction completes successfully, but no NFT is created.  
similar to what is happening on https://github.com/kodadot/nft-gallery/pull/6245#issuecomment-1597423277

https://kusama.subscan.io/tx/0xdfa6a4a41f4adaaf6b23c3bad425aea45d0259fe6721366c0839fd728269c046

image

**Comment by daiagi** [comment]
@prury

Sure. I'll check

**Comment by daiagi** [comment]
@prury 
Mint w royalty/ w/o listing:

image


mint w/o royalty w/o listing:

image

mint 3 copies, w royalties, w properties w/o listing:

image


mint 2 copies, w list (list would have appeared in the next transaction, but i cancel the minting)

image

and your own successful mint without listing as well:
https://kusama.subscan.io/extrinsic/18439604-4
https://deploy-preview-6156--koda-canary.netlify.app/ksm/gallery/18439604-7EA1DCF47E98A25067-PNKTST-RADIOACTIVE_PUNK_1-00000006

**Comment by prury** [comment]
Thank you for the many tests! 

> and your own successful mint without listing as well: https://kusama.subscan.io/extrinsic/18439604-4 https://deploy-preview-6156--koda-canary.netlify.app/ksm/gallery/18439604-7EA1DCF47E98A25067-PNKTST-RADIOACTIVE_PUNK_1-00000006

This one i minted on canary😛 

did another one today and it went well: 
https://kusama.subscan.io/tx/0xd5d1735f13d4aa784fe79fd642a49eedc4b3e3931e3a0e6fec7b12ae317dc76f

@daiagi last question, will page redirection after minting be done in another PR? otherwise works for me

**Comment by vikiival** [comment]
please resolve conflicts

**Comment by vikiival** [comment]
https://deploy-preview-6156--koda-canary.netlify.app/ksm/gallery/18453986-8cc1b91e899d9be40e-WAIF-TRAVEL_WAIFU_2-00000008

**Review by vikiival — APPROVED** [review]

**Pull request merged**